File: app/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/activities/fragments/investigation/evidence/children/solo/views/MapCarouselView.java
package com.TritiumGaming.phasmophobiaevidencepicker.activities.fragments.investigation.evidence.children.solo.views;

import androidx.appcompat.widget.AppCompatImageButton;
import androidx.appcompat.widget.AppCompatTextView;

import com.TritiumGaming.phasmophobiaevidencepicker.activities.fragments.investigation.evidence.children.solo.data.MapCarouselData;

/**
 * MapSelectControl class
 *
 * @author TritiumGamingStudios
 */
public class MapCarouselView {

    private final MapCarouselData mapCarouselData;

    private AppCompatTextView mapNameView = null;

    /**
     * MapSelectControl parameterized constructor
     *
     * @param prev
     * @param next
     * @param mapNameView
     */
    public MapCarouselView(
            MapCarouselData mapsCarouselData,
            AppCompatImageButton prev,
            AppCompatImageButton next,
            AppCompatTextView mapNameView) {

        this.mapCarouselData = mapsCarouselData;

        setDifficultyDisplay(mapNameView);
        setPrev(prev);
        setNext(next);
    }

    /**
     * setPrev method
     *
     * @param prev
     */
    private void setPrev(AppCompatImageButton prev) {
        prev.setOnClickListener(v -> {

            if (mapCarouselData != null) {
                int i = mapCarouselData.getMapCurrentIndex() - 1;
                if (i < 0) {
                    i = mapCarouselData.getMapCount() - 1;
                }
                setCurrentMapIndex(i);
            }

        });
    }

    /**
     * setNext method
     *
     * @param next
     */
    private void setNext(AppCompatImageButton next) {
        next.setOnClickListener(v -> {
            if (mapCarouselData != null) {
                int i = mapCarouselData.getMapCurrentIndex() + 1;
                if (i >= mapCarouselData.getMapCount()) {
                    i = 0;
                }
                setCurrentMapIndex(i);
            }
        });
    }

    private void setCurrentMapIndex(int i) {
        mapCarouselData.setMapCurrentIndex(i);
        mapNameView.setText(mapCarouselData.getMapCurrentName().split(" ")[0]);
    }

    /**
     * setDifficultyDisplay method
     *
     * @param display
     */
    private void setDifficultyDisplay(AppCompatTextView display) {
        this.mapNameView = display;
    }

}

File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/activities/investigation/evidence/views/MapCarouselView.kt
package com.TritiumGaming.phasmophobiaevidencepicker.activities.investigation.evidence.views

import android.view.View
import androidx.appcompat.widget.AppCompatImageButton
import androidx.appcompat.widget.AppCompatTextView
import com.TritiumGaming.phasmophobiaevidencepicker.activities.investigation.evidence.data.carousels.MapCarouselModel

/**
 * MapSelectControl class
 *
 * @author TritiumGamingStudios
 */
class MapCarouselView(
    private val mapCarouselData: MapCarouselModel?,
    prev: AppCompatImageButton,
    next: AppCompatImageButton,
    mapNameView: AppCompatTextView
) {
    private var mapNameView: AppCompatTextView? = null

    /**
     * MapSelectControl parameterized constructor
     *
     * @param prev
     * @param next
     * @param mapNameView
     */
    init {
        setDifficultyDisplay(mapNameView)
        setPrev(prev)
        setNext(next)
    }

    /**
     * setPrev method
     *
     * @param prev
     */
    private fun setPrev(prev: AppCompatImageButton) {
        prev.setOnClickListener {
            if (mapCarouselData != null) {
                var i = mapCarouselData.mapCurrentIndex - 1
                if (i < 0) {
                    i = mapCarouselData.mapCount - 1
                }
                setCurrentMapIndex(i)
            }
        }
    }

    /**
     * setNext method
     *
     * @param next
     */
    private fun setNext(next: AppCompatImageButton) {
        next.setOnClickListener {
            if (mapCarouselData != null) {
                var i = mapCarouselData.mapCurrentIndex + 1
                if (i >= mapCarouselData.mapCount) {
                    i = 0
                }
                setCurrentMapIndex(i)
            }
        }
    }

    private fun setCurrentMapIndex(i: Int) {
        mapCarouselData?.mapCurrentIndex = i
        mapNameView?.text =
            mapCarouselData?.mapCurrentName?.split(" ".toRegex())?.dropLastWhile { it.isEmpty() }
                ?.toTypedArray()?.get(0)
    }

    /**
     * setDifficultyDisplay method
     *
     * @param display
     */
    private fun setDifficultyDisplay(display: AppCompatTextView) {
        this.mapNameView = display
    }
}

